activerecord/test/cases/adapters/postgresql/postgresql_adapter_prevent_writes_test.rb

```diff
@@ -49,6 +49,16 @@ def test_errors_when_a_delete_query_is_called_while_preventing_writes
         end
       end
 
+      def test_errors_when_copy_from_is_used_while_preventing_writes
+        with_example_table do
+          ActiveRecord::Base.while_preventing_writes do
+            assert_raises(ActiveRecord::ReadOnlyError) do
+              @connection.execute("COPY ex (id) FROM STDIN")
```

Comment:
That regexp seems super naive. Following query seems pass and is considered read-only.

```sql
COPY ex (to) FROM STDIN;
```

Reply:
oh good catch! I am going to try to see I can tighten up that regex based on the definition of `COPY` 

```
COPY { table_name [ ( column_name [, ...] ) ] | ( query ) }
    TO { 'filename' | PROGRAM 'command' | STDOUT }
    [ [ WITH ] ( option [, ...] ) ]
```

Reply:
what do you think of 
```
copy\s+(.*)\s+to\s+(.*)
```
this passes correctly does not match with your example now and also some other variations I tested with.

Reply:
Maybe it makes sense to add the confusing cases to the tests, rather than the simple ones (or in addition to?)

Just to have confidence that the regex does what's expected in the edge cases